code/modules/clothing/spacesuits/rig/modules/specific/ai_container.dm

```diff
@@ -175,41 +184,210 @@
+	set name = "Enable Direct Control"
+	set desc = "Engage a neural jack."
+
+	var/mob/living/jacker = integrated_ai
+	var/obj/item/weapon/rig/rig = holder
+	var/mob/living/carbon/human/H = rig.wearer
+	if(!(istype(H) && H.get_rig() == rig))
+		to_chat(jacker, SPAN_WARNING("Your rig does not have a pilot, or is attached to an incompatible bioform."))
+		return
+
+	if(jacker.stat)
+		to_chat(src, SPAN_WARNING("You cannot do that in your current state."))
+		return
+
+	if(!holder.ai_override_enabled)
+		to_chat(src, SPAN_WARNING("You are locked out of \the [src]'s neural jack controls."))
+		return
+
+	to_chat(jacker, SPAN_NOTICE("You begin the process of enabling \the [src]'s neural jack."))
+	to_chat(H, SPAN_WARNING("\The [rig]'s [src] begins clacking, accompanied by a pressure along your spine."))
+
+	if(do_after(jacker, 1 MINUTE, H, ignore_movement = TRUE))
+		if(!H || !integrated_ai || controlling || !rig.ai_override_enabled)
+			return
+
+		to_chat(jacker, SPAN_WARNING("You fully engage \the [src]'s neural jack, interfacing directly with the pilot's nervous system."))
+		var/obj/item/organ/internal/brain/Brain = H.internal_organs_by_name[O_BRAIN]
+		if(!Brain)
+			to_chat(jacker, SPAN_WARNING("\The [src] responds with an error code, as the neural jack swiftly retracts. The pilot has no neural cortex."))
+			return
+
+		if(H.has_brain_worms())
+			to_chat(jacker, SPAN_WARNING("\The [src] connects, but only for a moment. You see the world through.. alien eyes, for a brief instant, as the jack retracts."))
+			to_chat(H, SPAN_WARNING("Something moves under the surface of your neck."))
+			return
+
+		if(Brain.robotic < ORGAN_ASSISTED)
+			to_chat(H, SPAN_WARNING("You feel a strange shifting sensation behind your eyes as a consciousness displaces yours. Your final sensation is that of something wet tearing."))
+			H.adjustBrainLoss(rand(0,5))
+			imperfect_connection = TRUE
+
+		else
+			to_chat(H, SPAN_WARNING("You feel a strange shifting sensation behind your eyes as a consciousness displaces yours."))
+
+		if(H.stat == DEAD)
+			corpse_pilot = TRUE
+
+		// pilot -> brain
+		qdel(pilot_brain)
+		pilot_brain = new(src)
+
+		H.transfer_player(pilot_brain)
+
+		// AI -> pilot
+		jacker.transfer_player(H)
+		controlling = TRUE
+		ai_locked = TRUE
+
+		H.verbs += /mob/living/carbon/proc/release_control
+
+		return
+
+/obj/item/rig_module/ai_container/advanced/proc/revert()
+	if(controlling && integrated_ai && pilot_brain)
+		var/mob/living/carbon/human/H = holder.loc	// The mob wearing the rig
+
+		if(istype(H))
+			controlling = FALSE
+			H.verbs -= /mob/living/carbon/proc/release_control
+
+			H.transfer_player(integrated_ai)
+
+			to_chat(pilot_brain, "<span class='notice'>Your mind returns from a strange limbo, the sensations of the body returning.[imperfect_connection ? " Your voice returns shakily." : ""][corpse_pilot ? " <span class='warning'>And then everything comes crashing down.</span>" : ""]</span>")
+
+			pilot_brain.transfer_player(H)
```

Comment:
What happens if the player ghosts out of the pilot brain?

Reply:
The pilot brain is the same one the borer uses, so it would be the same fate. They can resist to return to their body, and given this is unlikely to be used on unwilling individuals due to how easy it is , they probably won't do so out of frustration.